Client/Frontend/Browser/BackForwardTableViewCell.swift

```diff
@@ -0,0 +1,149 @@
+                bg.snp_updateConstraints { make in
+                    make.height.equalTo(BackForwardViewCellUX.faviconWidth+BackForwardViewCellUX.borderBold)
+                    make.width.equalTo(BackForwardViewCellUX.faviconWidth+BackForwardViewCellUX.borderBold)
+                }
+            }
+        }
+    }
+    
+    var site: Site? {
+        didSet {
+            if let s = site {
+                faviconView.setIcon(s.icon, withPlaceholder: FaviconFetcher.defaultFavicon)
+                label.text = s.title
+                setNeedsLayout()
+            }
+        }
+    }
+    
+    init(style: UITableViewCellStyle, reuseIdentifier: String?, isPrivate: Bool) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        backgroundColor = UIColor.clearColor()
+        selectionStyle = .None
+        
+        let selectedView = UIView()
+        selectedView.backgroundColor = UIColor.redColor()
+        selectedBackgroundView =  selectedView;
+        
+        faviconView = UIImageView(image: FaviconFetcher.defaultFavicon)
+        faviconView.backgroundColor = UIColor.whiteColor()
+        contentView.addSubview(faviconView)
+        
+        label = UILabel(frame: CGRectZero)
+        if isPrivate {
+            label.textColor = UIColor.whiteColor()
+        }
+        else {
+            label.textColor = UIColor.blackColor()
+        }
+        
+        label.text = " "
+        label.font = label.font.fontWithSize(BackForwardViewCellUX.fontSize)
+        contentView.addSubview(label)
+        
+        faviconView.snp_makeConstraints { make in
+            make.height.equalTo(BackForwardViewCellUX.faviconWidth)
+            make.width.equalTo(BackForwardViewCellUX.faviconWidth)
+            make.centerY.equalTo(self)
+            make.left.equalTo(self.snp_left).offset(BackForwardViewCellUX.faviconPadding)
+        }
+        
+        label.snp_makeConstraints { make in
+            make.centerY.equalTo(self)
+            make.left.equalTo(faviconView.snp_right).offset(BackForwardViewCellUX.labelPadding)
+            make.right.equalTo(self.snp_right).offset(BackForwardViewCellUX.labelPadding)
+        }
+        
+        bg = UIView(frame: CGRect.zero)
+        bg.backgroundColor = BackForwardViewCellUX.bgColor
+        
+        self.addSubview(bg)
+        self.sendSubviewToBack(bg)
+        
+        bg.snp_makeConstraints { make in
+            make.height.equalTo(BackForwardViewCellUX.faviconWidth+BackForwardViewCellUX.borderSmall)
+            make.width.equalTo(BackForwardViewCellUX.faviconWidth+BackForwardViewCellUX.borderSmall)
+            make.centerX.equalTo(faviconView)
+            make.centerY.equalTo(faviconView)
+        }
+        
+    }
+    
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func drawRect(rect: CGRect) {
+        let context = UIGraphicsGetCurrentContext();
```

Comment:
Whenever overriding framework methods like this, I always try to call the super method to make sure the framework does what it needs to do first.